jobs/delorean/jenkinsfiles/branch/suite-next/Jenkinsfile

```diff
@@ -0,0 +1,112 @@
+#!groovy
+
+// Ensure required parameters are provided
+def verifyParameters(suiteParams) {
+  if (!suiteParams.gitUrl || !suiteParams.suiteBranch) {
+      def userInput = input message: 'Suites Next Options', parameters: [
+        string(defaultValue: (suiteParams.gitUrl ?: 'git@github.com:integr8ly/installation.git'), description: 'The installation repo git url', name: 'installationGitUrl'),
+        string(defaultValue: (suiteParams.suiteBranch ?: ''), description: 'Desired name for the suite branch', name: 'suiteBranch'),
+      ]
+      suiteParams.gitUrl = userInput.installationGitUrl
+      suiteParams.suiteBranch = userInput.suiteBranch
+      verifyParameters(suiteParams)
+  }
+}
+
+// Checkout remote repository
+def checkoutGitRepo(gitUrl, gitRef, credentialsID) {
+  checkout([$class: 'GitSCM', branches: [[name: gitRef]],
+            doGenerateSubmoduleConfigurations: false,
+            extensions: [],
+            submoduleCfg: [],
+            userRemoteConfigs: [[credentialsId: credentialsID, url: gitUrl]]])
+}
+
+// Get product next branches available in the installation repo
+def getProductNextBranches(gitOrg, gitRepo, sourceBranch) {
+  nextBranches = sh(returnStdout: true, script: "git ls-remote origin | grep next | awk '{print \$2}'").replaceAll("refs/heads/", "").split()
+
+  // Do not include the source branch in the next branches list
+  productNextBranches = nextBranches.minus(sourceBranch)
+  return productNextBranches
+}
```

Comment:
There shouldn't be any need to deal with product next branches in here, this Jenkinsfile is specific to suite testing i.e. Transient branches that will never be merged back or have PRs opened for them, and product branches being meregd shoudl be passed in explicitly. We have a separate Jenkinsfile for dealing with the product-next branch flow  https://github.com/integr8ly/ci-cd/blob/master/jobs/delorean/jenkinsfiles/branch/next/Jenkinsfile